readthedocs/search/faceted_search.py

```diff
@@ -357,13 +357,26 @@ def _get_nested_query(self, *, query, path, fields):
             query=query,
             fields=fields,
         )
+        bool_query = Bool(should=queries)
 
         raw_fields = [
             # Remove boosting from the field
             re.sub(r'\^.*$', '', field)
             for field in fields
         ]
 
+        # The ``post_filter`` filter will only filter documents
+        # at the parent level (domains is a nested document),
+        # resulting in results with domains that don't match the current
+        # role_name being filtered, so we need to force filtering by role_name
+        # on the ``domains`` document here. See #8268.
+        # TODO: We should use a flattened document instead
+        # to avoid this kind of problems and have faster queries.
```

Comment:
Yea, I feel like we've mentioned this a number of times. Really should figure it out at some point. Is this tracked in an issue or roadmap card anywhere?

Reply:
I'll create a new issue and add it for a future task in our roadmap :+1:

Reply:
Done https://github.com/readthedocs/readthedocs.org/issues/8457, I put a card on the "Needs discussion" column.